### ISSUE TITLE
dynamic articon icon pack doesnt changes 

### ISSUE THREAD

**Issue by ansarimohammed422** [MM2-0/Kvaesitso#519]
It doesn't changes when changing wallpaper

I have turned on dynamic icon setting in icon section in launcher

The launcher colour changes

It is not working from this latest release

But it was working fine on previous release

**Comment by Crazy-Unicooorn** [comment]
Might be related: the toggle button to change the theme of the icon pack doesn't work either after 1.17

![Screenshot_20230828_233025_Kvaesitso.jpg](https://github.com/MM2-0/Kvaesitso/assets/30509883/64564bb6-4f83-4919-baa6-a22a61608bb9)

This button on the right on the screenshot.
Previously it was changing how the icons look (I think kind of a dark/light mode toggle), now it just doesn't do anything.

**Comment by MM2-0** [comment]
I can't reproduce either issue. What Androids are you using?

**Comment by ansarimohammed422** [comment]
Android 13 Miui 14

**Comment by Crazy-Unicooorn** [comment]
Android 13
One Ui 5.1

Here is a video of my issue:

https://github.com/MM2-0/Kvaesitso/assets/30509883/2f2e1405-18f8-4150-bb85-dffb26cae7db

**Comment by exstntlmsnthrp** [comment]
I'm using LineageOS 20 (Android 13) on Pixel 7 Pro.

**Comment by exstntlmsnthrp** [comment]
> Might be related: the toggle button to change the theme of the icon pack doesn't work either after 1.17
> 
> ![Screenshot_20230828_233025_Kvaesitso.jpg](https://github.com/MM2-0/Kvaesitso/assets/30509883/64564bb6-4f83-4919-baa6-a22a61608bb9)
> 
> This button on the right on the screenshot.
> Previously it was changing how the icons look (I think kind of a dark/light mode toggle), now it just doesn't do anything.

I suspect what you reference here is more related to the icon pack and not kvaesitso. The button in your screenshot works with Arcticons dark or light. I can confirm it does not work with Arcticons You, but I believe that is the intended behaviour as they are material you and should change automatically.

**Comment by Crazy-Unicooorn** [comment]
> > Might be related: the toggle button to change the theme of the icon pack doesn't work either after 1.17
> > 
> > ![Screenshot_20230828_233025_Kvaesitso.jpg](https://github.com/MM2-0/Kvaesitso/assets/30509883/64564bb6-4f83-4919-baa6-a22a61608bb9)
> > 
> > This button on the right on the screenshot.
> > Previously it was changing how the icons look (I think kind of a dark/light mode toggle), now it just doesn't do anything.
> 
> @Crazy-Unicooorn I suspect what you reference here is more related to the icon pack and not kvaesitso. The button in your screenshot works with Arcticons dark or light. I can confirm it does not work with Arcticons You, but I believe that is the intended behaviour as they are material you and should change automatically. 

I don't think so, because before the newest release it was changing how the icons look. You could switch so that instead of the accent color being on the background it was on the lines of the icons.

I know it for a fact because what happened is that I was using the other style, I updated Kvaesitso, it changed all my icons, I tried putting the other style back by pressing the toggle, and that's how I discovered there is a bug, as it didn't do anything.

**Comment by exstntlmsnthrp** [comment]
I'm using LineageOS 20 (Android 13) on Pixel 7 Pro.

**Comment by Crazy-Unicooorn** [comment]
I see. Then I don't know why I was able to get this, but here is what my icons looked like before the update:

![Screenshot_20230830_131809_TikTok.jpg](https://github.com/MM2-0/Kvaesitso/assets/30509883/4c4a8713-5d10-4ca5-990d-705e42f42fcc)

Here is what they look like now:

![Screenshot_20230830_131844_TikTok.jpg](https://github.com/MM2-0/Kvaesitso/assets/30509883/bc535341-7fe2-4985-aad4-a2042e77c53c)

And I clearly remember playing with this toggle when it was working, to choose which version I liked better.

**Comment by exstntlmsnthrp** [comment]
I see what you say. I didn't doubt you, but agree that is rather odd. Whilst I have nothing to back this up, I wonder if it has to do with recent updates to the icon packs for Arcticons?

**Comment by Crazy-Unicooorn** [comment]
Sure no worries! I'm glad that you commented because it allows me to be more specific and share more details.
Now I just hope I am not polluting this issue with something unrelated.

> I wonder if it has to do with recent updates to the icon packs for Arcticons?
I don't know, that might be the case yes! But I am quite sure I did not do both updates at the same time.

I will open an issue to the Arcticons repo, because the toggle does work for me with the other Arcticons icon packs (I just tried that).
So it indeed sounds related to Arcticons You itself, which is weird to me as I remember being excited about updating Kvaesitso, going back to the homescreen after completion, and seeing my icons changed.

**Comment by MM2-0** [comment]
I have changed the colors of themed icons, I've reverted them to pre 1.26. Maybe that's the issue here?

**Comment by Donnnno** [comment]
Hi! Here's a quick explanation on how Articons You works:

- All lines use a color like system_accent_100 (idk the actual value atm), with a background that has like the opposite value.

On devices that run on android 11 and lower, it switches to black and white for both values.

This seems like a launcher / system bug to me

**Comment by Donnnno** [comment]
My suggestion would be to check if nova or smart launcher have this issue too. Those implementations are probably the most reliable

**Comment by MM2-0** [comment]
Oh, I see it now. I didn't read the "You" part. You are right, Arcticons You icons aren't themed properly. The other Arcticons packs work fine though.

>Hi! Here's a quick explanation on how Articons You works:
>    All lines use a color like system_accent_100 (idk the actual value atm), with a background that has like the opposite value.
> On devices that run on android 11 and lower, it switches to black and white for both values.
> This seems like a launcher / system bug to me

Here is what the launcher does (or at least what it is supposed to do) when you enable that "dynamic colors" toggle button:

- If the icon is an adaptive icon, and the drawable has a <monochrome> layer, the monochrome layer is used as the foreground layer.
- If the icon is an adaptive icon, and it does not have a <monochrome> layer, the foreground layer is used.
- If the icon is not an adaptive icon, the entire icon is used.

In any case, the foreground layer is tinted with a color from the launchers color scheme (onPrimaryContainer) and the background is set to a solid color (primaryContainer)

This is different from what Archticons You does in launchers without dedicated themed icons support since
- it uses the actual launcher color scheme and light/dark mode preferenc, which can be different from the system
- it works on pre-Android 12

If you don't enable the toggle, the icons are used as is.

**Comment by exstntlmsnthrp** [comment]
Thanks for the explanation @MM2-0. That makes more sense.

**Comment by ansarimohammed422** [comment]
I have found a fix for this issue

When we enable "Enforce themed icon" in settings the icon pack gets fixed and works dynamically.

But when will a permenant fix will be their

**Comment by texastoland** [comment]
I'm using an adaptive icon pack. It uses my You color palette but doesn't automatically switch when dark mode is toggled. Would that be a separate issue? Or is that working as intended?
